### PR TITLE
Exposing the choice of train_sampler and validation_sampler for MQCNN and MQRNN 

### LINKED ISSUE #2380: Different lengths of training time series giving same model performance for MQCNN and MQRNN
## Description
(A clear and concise description of what the bug is.)
For the MQCNN and MQRNN models, giving more training data -- by extending the time series backwards in time -- does not result in any change in the performance, as seen by the `avg_epoch_loss` printout. If the whole training time series is used we would expect these to be different.

## To Reproduce
(Please provide minimal example of code snippet that reproduces the error. For existing examples, please provide link.)

```python
from gluonts.mx.model.seq2seq import MQCNNEstimator
from gluonts.mx import Trainer
from gluonts.dataset.repository.datasets import get_dataset

# Set up two training time series, one long, and one short (200 points)
dataset = get_dataset('electricity')
dataset_list = list(dataset.train)

longer = dataset_list[0]
shorter = {
    'target': longer['target'][-200:],
    'start': longer['start'] + len(dataset_list[0]['target']) - 200,
    'feat_static_cat': longer['feat_static_cat'],
    'item_id': longer['item_id']
}

seed = 3

for train_data in [shorter, longer]:
                
    trainer = Trainer(epochs=2)
    mq_cnn = MQCNNEstimator(prediction_length=24, 
                            freq='H', 
                            trainer=trainer,
                            seed=seed)
    print('Length: %s, \t first ten entries: %s' %(
        len(train_data['target']), train_data['target'][0:10]))
    model = mq_cnn.train(training_data=[train_data], )
```

## Error message or code output
(Paste the complete error message, including stack trace, or the undesired output that the above snippet produces.)

```
Length: 200, 	 first ten entries: [2. 3. 2. 2. 0. 4. 2. 4. 2. 8.]

100%|██████████████████████████████████| 50/50 [00:05<00:00,  8.41it/s, epoch=1/2, avg_epoch_loss=0.476]
100%|██████████████████████████████████| 50/50 [00:05<00:00,  8.64it/s, epoch=2/2, avg_epoch_loss=0.298]

Length: 21044, 	 first ten entries: [14. 18. 21. 20. 22. 20. 20. 20. 13. 11.]

100%|██████████████████████████████████| 50/50 [00:12<00:00,  4.00it/s, epoch=1/2, avg_epoch_loss=0.476]
100%|██████████████████████████████████| 50/50 [00:12<00:00,  4.01it/s, epoch=2/2, avg_epoch_loss=0.298]
```

## Environment
- Operating system: Mac OS Monterey
- Python version: 3.8.13
- GluonTS version: 0.10.6
- MXNet version: 1.9.1

### PR DESCRIPTION
**Pull request by sighellan**
Fixes: https://github.com/awslabs/gluonts/issues/2380

*Description of changes:*
- Lets the user specify what sampler they want to use for the MQCNN and MQRNN models, keeping the default as before. 
- Adding comment to docstring describing default behaviour.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** other change

### CODE REVIEW

File: src/gluonts/mx/model/seq2seq/_mq_dnn_estimator.py [hunk 1]
```diff
@@ -135,6 +140,9 @@ class MQCNNEstimator(ForkingSeq2SeqEstimator):
         Determines whether to use IQF or QF. (default: True).
     batch_size
         The size of the batches to be used training and prediction.
+    train_sampler
+        Controls the sampling of windows during training.
```
Comment (lostella) [inline src/gluonts/mx/model/seq2seq/_mq_dnn_estimator.py:144]: Maybe it makes sense to similarly expose the `validation_sampler` that the upper class exposes

**Review by lostella — CHANGES_REQUESTED** [review]

File: src/gluonts/mx/model/seq2seq/_mq_dnn_estimator.py (hunk 1 above)
Comment (sighellan) [reply src/gluonts/mx/model/seq2seq/_mq_dnn_estimator.py:144]: The validation_sampler is also exposed now.

**Comment by sighellan** [comment]
**Motivation for the error message update:** 

I tested the MQRNN model using `ExpectedNumInstanceSampler` for the `validation_sampler`, which resulted in failures for some but not other seeds. This could be fixed by increasing the `num_instances` parameter. 

Code: 
```
from gluonts.transform import ExpectedNumInstanceSampler
from gluonts.mx.model.seq2seq import MQRNNEstimator
from gluonts.mx import Trainer
from gluonts.dataset.repository.datasets import get_dataset
import numpy as np
import mxnet as mx

def make_shorter(tseries, ll=500):
    shorter = {
        'target': tseries['target'][-ll:],
        'start': tseries['start'] + len(tseries['target']) - ll,
        'feat_static_cat': tseries['feat_static_cat'],
        'item_id': tseries['item_id']
    }
    return shorter

dataset = get_dataset('electricity')
shorter = make_shorter(list(dataset.train)[0])
shorter_val = make_shorter(list(dataset.test)[0])

for val_num_instances in [1, 10]:
    successes = 0
    failures = 0
    print('----------------------')
    print('Validation num_instances: %s' %val_num_instances)
    for seed in [0, 1, 0, 1]:
        try:

            trainer = Trainer(epochs=1)
            np.random.seed(seed)
            mx.random.seed(seed, trainer.ctx)

            estimator_new = MQRNNEstimator(prediction_length=24, 
                                           freq='H', 
                                           trainer=trainer,
                                           train_sampler=ExpectedNumInstanceSampler(
                                           num_instances=1, min_future=24),
                                           validation_sampler=ExpectedNumInstanceSampler(
                                           num_instances=val_num_instances, 
                                               min_future=24)) 

            model_new = estimator_new.train(training_data=[shorter], 
                                            validation_data=[shorter_val])
            print('Success!')
            successes += 1
        except Exception as e:
            print(e)
            print('Failure!')
            failures += 1

    print('Successes: %s' %successes)
    print('Failures: %s' %failures)
```

Previous output: 
```
----------------------
Validation num_instances: 1

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.63it/s, epoch=1/1, avg_epoch_loss=1.85]
0it [00:00, ?it/s]

No training data batch could be constructed; this usually indicates that the training dataset is empty, or consists of too short series.
Failure!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.54it/s, epoch=1/1, avg_epoch_loss=1.88]
1it [00:00, 139.68it/s, epoch=1/1, validation_avg_epoch_loss=1.65]

Success!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.64it/s, epoch=1/1, avg_epoch_loss=1.85]
0it [00:00, ?it/s]

No training data batch could be constructed; this usually indicates that the training dataset is empty, or consists of too short series.
Failure!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.52it/s, epoch=1/1, avg_epoch_loss=1.88]
1it [00:00, 143.56it/s, epoch=1/1, validation_avg_epoch_loss=1.65]

Success!
Successes: 2
Failures: 2
----------------------
Validation num_instances: 10

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.57it/s, epoch=1/1, avg_epoch_loss=1.85]
1it [00:00, 29.02it/s, epoch=1/1, validation_avg_epoch_loss=1.67]

Success!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.57it/s, epoch=1/1, avg_epoch_loss=1.88]
1it [00:00, 30.84it/s, epoch=1/1, validation_avg_epoch_loss=1.54]

Success!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.67it/s, epoch=1/1, avg_epoch_loss=1.85]
1it [00:00, 30.93it/s, epoch=1/1, validation_avg_epoch_loss=1.67]

Success!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.60it/s, epoch=1/1, avg_epoch_loss=1.88]
1it [00:00, 29.35it/s, epoch=1/1, validation_avg_epoch_loss=1.54]

Success!
Successes: 4
Failures: 0
```

New output with more informative error message:
```
----------------------
Validation num_instances: 1

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.63it/s, epoch=1/1, avg_epoch_loss=1.85]
0it [00:00, ?it/s]

No validation data batch could be constructed; this usually indicates that the validation dataset is empty, or consists of too short series. If using a random data sampler, this might be caused by not taking enough samples.
Failure!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.65it/s, epoch=1/1, avg_epoch_loss=1.88]
1it [00:00, 123.79it/s, epoch=1/1, validation_avg_epoch_loss=1.65]

Success!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.51it/s, epoch=1/1, avg_epoch_loss=1.85]
0it [00:00, ?it/s]

No validation data batch could be constructed; this usually indicates that the validation dataset is empty, or consists of too short series. If using a random data sampler, this might be caused by not taking enough samples.
Failure!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.68it/s, epoch=1/1, avg_epoch_loss=1.88]
1it [00:00, 139.82it/s, epoch=1/1, validation_avg_epoch_loss=1.65]

Success!
Successes: 2
Failures: 2
----------------------
Validation num_instances: 10

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.59it/s, epoch=1/1, avg_epoch_loss=1.85]
1it [00:00, 30.03it/s, epoch=1/1, validation_avg_epoch_loss=1.67]

Success!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.58it/s, epoch=1/1, avg_epoch_loss=1.88]
1it [00:00, 29.19it/s, epoch=1/1, validation_avg_epoch_loss=1.54]

Success!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.60it/s, epoch=1/1, avg_epoch_loss=1.85]
1it [00:00, 30.40it/s, epoch=1/1, validation_avg_epoch_loss=1.67]

Success!

100%|████████████████████████████████████████████████████████████████| 50/50 [00:07<00:00,  6.56it/s, epoch=1/1, avg_epoch_loss=1.88]
1it [00:00, 29.65it/s, epoch=1/1, validation_avg_epoch_loss=1.54]

Success!
Successes: 4
Failures: 0
```

**Review by lostella — APPROVED** [review]
Thanks @sighellan! Looks good!
